README.md

```diff
@@ -86,6 +86,6 @@ When specifying multiple labels, you will need to repeat the argument name, e.g:
 * PromQL supports queries without `__name__`. This is not possible in OpenTSDB and no results will be returned if the query doesn't match on a metric name.
 * Geras periodically loads metric names from OpenTSDB and keeps them in memory to support queries like `{__name__=~"regexp"}`.
 * Thanos' primary timeseries backend is Prometheus, which doesn't support unquoted dots in metric names. However OpenTSDB metrics generally use `.` as a seperator within names. In order to query names containing a `.` you will need to either:
-  * PromQL's [data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels) requires many characters not exist
-    * We use the following match on metric names to obey that model: `regexp.MustCompile("[^a-zA-Z0-9_:]")`
+  * Replace all `.` with another character (we like `:`).
```

Comment:
The `:` is still hardcoded on the query side, it's this one: https://github.com/G-Research/geras/blob/master/pkg/store/store.go#L380

It maybe makes sense to use periodCharacter here though for consistency, even if I don't think anything other than `:` makes sense.

I've spent too long thinking about this, `:` is just the best default compromise we have. (`_` would match much better with how Prometheus uses it, but the problem is, if you were to change all `_` to `.`, then it wouldn't be possible to query a metric with an `_` in it at all).

Also please update the flags listed above.

Reply:
For our use case, because we'd started doing filtering on ingest into our new store that strictly follows the data model (and moves everything to `_`), being able to make queries from OpenTSDB come back the same way is nice. So that's _why_ I did it.
I'm not actually sure how to run the test cases?

Reply:
Ok, that makes more sense.

You can run the tests like this:
```
$ go test ./...       
# github.com/G-Research/geras/cmd/geras
cmd/geras/main.go:189:31: too many arguments in call to store.NewOpenTSDBStore
	have ("github.com/go-kit/kit/log".Logger, client.ClientContext, prometheus.Registerer, time.Duration, time.Duration, []storepb.Label, *regexp.Regexp, *regexp.Regexp, bool, bool, string, string)
	want ("github.com/go-kit/kit/log".Logger, client.ClientContext, prometheus.Registerer, time.Duration, time.Duration, []storepb.Label, *regexp.Regexp, *regexp.Regexp, bool, bool, string)
ok  	github.com/G-Research/geras/pkg/regexputil	0.002s
--- FAIL: TestComposeOpenTSDBQuery (0.00s)
    store_test.go:22: {"component":"opentsdb","level":"debug","tsdb-query":"{\"start\":0,\"end\":100,\"queries\":[{\"aggregator\":\"none\",\"metric\":\".t.e.s.t...m.e.t.r.i.c.\",\"filters\":[{\"type\":\"literal_or\",\"tagk\":\"key\",\"filter\":\"value\",\"groupBy\":true}]}],\"msResolution\":true,\"showQuery\":true}"}
```

(or just `go test` if you're working in a package's directory)

Reply:
```
$ go test ./cmd/geras/
?   	github.com/G-Research/geras/cmd/geras	[no test files]
```
It worked once, then started failing. Sorry I'm not great at tests...I'm still kinda learning Go.

Reply:
Either run it at the top level of the repo with `go test ./...`, or inside pkg/store with just `go test` -- there's not any tests for the binary on its own currently.

Reply:
Yep. As is the rules, the minute I commented, I realized what I was doing wrong. Cleaning up test runs now.

Reply:
Flummoxed by this one:
```
$ go test ./...
?   	github.com/G-Research/geras/cmd/geras	[no test files]
ok  	github.com/G-Research/geras/pkg/regexputil	(cached)
--- FAIL: TestGetMatchingMetricNames (0.00s)
    store_test.go:1360: Metric .m.e.t.r.i.c...n.a.m.e. doesn't match expected: metric.name
FAIL
FAIL	github.com/G-Research/geras/pkg/store	0.005s
FAIL
```
But it appears to be the only error left.

Reply:
It's because periodCharacter ends up empty, as https://github.com/G-Research/geras/blob/2af85eae6ca8a7c5c4f8c3c1d65930e029761b85/pkg/store/store_test.go#L1333 has its own OpenTSDBStore initialiser.

It does raise the slightly interesting question of whether a user should be able to specify an empty string for --period-character-replace as that would be confusing for a user too. Maybe empty should disable the rewriting, i.e. fix this by making the replacement not happen at https://github.com/G-Research/geras/pull/87/commits/db29b78c24044d8ab790570a42ae9a29fb3ab42c#diff-fb9f8e8a65e63ef4cf1e15404b8bc0aa9d3694209d9ba484984f6a78f2207b64R382 if it's empty.